Add check to prevent attaching a profiler when one is already present

davmason opened the pull request.

davmason commented:
Fixes #25313

noahfalk commented on src/vm/profilerdiagnosticprotocolhelper.cpp:
```diff
@@ -103,6 +103,14 @@ void ProfilerDiagnosticProtocolHelper::AttachProfiler(DiagnosticsIpc::IpcMessage
         goto ErrExit;
     }
     
+    // The diagnostic server is single threaded, so we do not have to worry about racing
```
I'd recommend making the ProtocolHelper be responsible only for marshalling serialized buffer -> arguments and return value -> serialized buffer. For any logic related to the operation itself I'd put it inside LoadProfilerForAttach.

davmason replied:
Yeah, I think that makes more sense

davmason commented:
@noahfalk I moved the check in to `ProfilingAPIUtility::LoadProfiler`, and it occurred to me that the attach path could race with the startup path since the diagnostics server starts so early. To solve this I used added `ProfilerStatus::kProfStatusPreInitialize` to keep track of if there is already a profiler trying to load. I moved a bunch of logic to a new function so I could aggregate any errors and just set it back to `kProfStatusNone` in one location, but the logic is unchanged.

davmason commented:
Also meant to mention I did manual verification that attach works and a second attach request is refused, and then ran the profiler tests against it.
